codec/collections.go

```
@@ -91,6 +103,139 @@ func (c collValue[T, PT]) ValueType() string {
+							dur := &durationpb.Duration{}
+							err = c.cdc.Unmarshal(msgbz, dur)
+							if err != nil {
+								return nil, fmt.Errorf("error unmarshalling duration: %w", err)
+							}
+							values = append(values, dur.AsDuration())
+						} else {
+							// if not a time or duration, just keep it as a JSON object
+							// we might want to change this to include the entire object as separate fields
+							bz, err := c.cdc.MarshalJSON(msg)
+							if err != nil {
+								return nil, fmt.Errorf("error marshalling message: %w", err)
```

golangci-lint (1.61.0)

`marshalling` is a misspelling of `marshaling`

(misspell)